lib/railway_ipc/commands_consumer.ex

```diff
@@ -44,9 +44,16 @@ defmodule RailwayIpc.CommandsConsumer do
           @stream_adapter.ack(channel, delivery_tag)
         end
 
-        publish_function = fn event ->
-          RailwayIpc.Publisher.publish(channel, exchange, event)
-        end
+        publish_function =
```

Comment:
This allows us to optionally configure a consumer to specify its publishing module  + function. This way, we can configured the `RepublishedMessagesConsumer` to use a version of the publish function that will take in a persisted published message record and use that record's exchange to enact a "re-publish". Not sure if there is a better way to go about this to avoid having to specify a different publish function for this code flow? Problem is that we don't know the exchange to which we are publishing until we have the persisted published message we want to re-publish. We could re-use the original `Publisher.publish/3` function but then we'd be responsible for looking up the connection's channel and the persisted published message's exchange and then passing them into the call to `publish/3`.